### PR TITLE
refactor queues and schemes

### PR DESCRIPTION
**Pull request by jellegerbrandy**
big refactor of schemes and proposal types

### CODE REVIEW

File: src/proposal.ts [hunk 1]
```diff
@@ -67,18 +73,20 @@ export interface IProposalState {
   queuedVotePeriodLimit: number // in seconds (?)
   preBoostedVotePeriodLimit: number
   limitExponentValue: number
+  minimumDaoBounty: BN // in GEN
   proposingRepReward: BN // in REP
   preBoostedAt: Date
   proposer: Address
-  queue: IQueueState
+  queue: IQueueState,
```
Comment (tibetsprague) [inline src/proposal.ts]: no comma

File: src/queue.ts [hunk 2]
```diff
@@ -1,76 +1,90 @@
...
+  public static search(options: {dao?: Address, name?: string}, context: Arc): Observable<Queue[]> {
     let where = ''
     for (const key of Object.keys(options)) {
       const value = (options as any)[key]
-      // querying by'name' will not be predicable as the name is not always populated
-      if (value !== undefined && value !== 'name') {
-        where += `${key}: "${value}"\n`
+      if (value !== undefined) {
+        if (key === 'dao')  {
+          where += `dao: "${value}"\n`
```
Comment (tibetsprague) [inline src/queue.ts]: isnt this exactly the same as the condition below?

File: src/queue.ts [hunk 3]
```diff
@@ -1,76 +1,90 @@
...
+    // `
+    const query = gql`{
+      controllerSchemes (where: {${where}}) {
+        id
+        dao { id }
+        name
+        address
+      }
+    }`
+    console.log(query.loc.source.body)
```
Comment (tibetsprague) [inline src/queue.ts]: console.log

File: src/queue.ts [hunk 4]
```diff
@@ -1,76 +1,90 @@
...
+  ) {
     this.context = context
-    this.id = id
-    this.dao = dao
-    this.name = name
-    this.scheme = scheme
   }
 
   public state(): Observable<IQueueState> {
+    //
```
Comment (tibetsprague) [inline src/queue.ts]: ?

File: src/queue.ts [hunk 5]
```diff
@@ -95,26 +109,25 @@ export class Queue {
...
-          id: this.id,
-          name: this.name,
-          scheme: this.scheme,
-          threshold: 1,
-          votingMachine: this.context.contractAddresses.GenesisProtocol
-        }
-      }
-
+    const itemMap = (item: any): IQueueState => {
+      console.log(item)
```
Comment (tibetsprague) [inline src/queue.ts]: console

File: src/scheme.ts [hunk 6]
```diff
@@ -1,80 +1,115 @@
...
+  constructor(id: Address, dao: Address, name: string, address: Address, public context: Arc) {
     this.context = context
+    this.id = id
+    this.dao = dao
+    this.name = name
+    this.address = address
   }
 
+  public state(): Observable<ISchemeState> {
+    //
```
Comment (tibetsprague) [inline src/scheme.ts]: ?

File: test/proposal-genericscheme.spec.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
 import BN = require('bn.js')
 import { Arc } from '../src/arc'
 import {
-  IGenericScheme,
   IProposalStage,
   IProposalState,
   IProposalType,
   Proposal
   } from '../src/proposal'
+import {   IGenericScheme } from '../src/schemes/genericScheme'
```
Comment (tibetsprague) [inline test/proposal-genericscheme.spec.ts]: weird spacing

**Comment by jellegerbrandy** [comment]
closing this, meged it to #225 and I'll continue there.

**Pull request closed without merging**